comparison between signed and unsigned integer expressions

```
[17:55:29]	In file included from /paddle/paddle/fluid/operators/detection_map_op.cc:15:0:
[17:55:29]	/paddle/paddle/fluid/operators/detection_map_op.h: In instantiation of 'void paddle::operators::DetectionMAPOpKernel<Place, T>::GetBoxes(const paddle::framework::LoDTensor&, const paddle::framework::LoDTensor&, std::vector<std::map<int, std::vector<paddle::operators::DetectionMAPOpKernel<Place, T>::Box> > >&, std::vector<std::map<int, std::vector<std::pair<T, paddle::operators::DetectionMAPOpKernel<Place, T>::Box> > > >&) const [with Place = paddle::platform::CPUPlace; T = double]':
[17:55:29]	/paddle/paddle/fluid/operators/detection_map_op.h:84:13:   required from 'void paddle::operators::DetectionMAPOpKernel<Place, T>::Compute(const paddle::framework::ExecutionContext&) const [with Place = paddle::platform::CPUPlace; T = double]'
[17:55:29]	/paddle/paddle/fluid/operators/detection_map_op.cc:184:67:   required from here
[17:55:29]	/paddle/paddle/fluid/operators/detection_map_op.h:154:38: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[17:55:29]	       for (int i = label_index[n]; i < label_index[n + 1]; ++i) {
[17:55:29]	                                      ^
[17:55:29]	/paddle/paddle/fluid/operators/detection_map_op.h:170:39: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[17:55:29]	       for (int i = detect_index[n]; i < detect_index[n + 1]; ++i) {
[17:55:29]	                                       ^
[17:55:29]	Scanning dependencies of target sequence_reshape_op
[17:55:29]	/paddle/paddle/fluid/operators/detection_map_op.h: In instantiation of 'paddle::operators::DetectionMAPOpKernel<Place, T>::GetInputPos(const paddle::framework::Tensor&, const paddle::framework::LoDTensor&, const paddle::framework::LoDTensor&, std::map<int, int>&, std::map<int, std::vector<std::pair<T, int> > >&, std::map<int, std::vector<std::pair<T, int> > >&) const::<lambda(const paddle::framework::LoDTensor&, std::map<int, std::vector<std::pair<T, int> > >&)> [with Place = paddle::platform::CPUPlace; T = double]':
[17:55:29]	/paddle/paddle/fluid/operators/detection_map_op.h:268:21:   required from 'struct paddle::operators::DetectionMAPOpKernel<Place, T>::GetInputPos(const paddle::framework::Tensor&, const paddle::framework::LoDTensor&, const paddle::framework::LoDTensor&, std::map<int, int>&, std::map<int, std::vector<std::pair<T, int> > >&, std::map<int, std::vector<std::pair<T, int> > >&) const [with Place = paddle::platform::CPUPlace; T = double]::<lambda(const class paddle::framework::LoDTensor&, class std::map<int, std::vector<std::pair<double, int>, std::allocator<std::pair<double, int> > >, std::less<int>, std::allocator<std::pair<const int, std::vector<std::pair<double, int>, std::allocator<std::pair<double, int> > > > > >&)>'
[17:55:29]	/paddle/paddle/fluid/operators/detection_map_op.h:280:5:   required from 'void paddle::operators::DetectionMAPOpKernel<Place, T>::GetInputPos(const paddle::framework::Tensor&, const paddle::framework::LoDTensor&, const paddle::framework::LoDTensor&, std::map<int, int>&, std::map<int, std::vector<std::pair<T, int> > >&, std::map<int, std::vector<std::pair<T, int> > >&) const [with Place = paddle::platform::CPUPlace; T = double]'
[17:55:29]	/paddle/paddle/fluid/operators/detection_map_op.h:91:18:   required from 'void paddle::operators::DetectionMAPOpKernel<Place, T>::Compute(const paddle::framework::ExecutionContext&) const [with Place = paddle::platform::CPUPlace; T = double]'
[17:55:29]	/paddle/paddle/fluid/operators/detection_map_op.cc:184:67:   required from here
[17:55:29]	/paddle/paddle/fluid/operators/detection_map_op.h:272:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[17:55:29]	       for (int i = 0; i < pos_data_lod.size(); ++i) {
[17:55:29]	                         ^
[17:55:29]	/paddle/paddle/fluid/operators/detection_map_op.h:273:44: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[17:55:29]	         for (int j = pos_data_lod[0][i]; j < pos_data_lod[0][i + 1]; ++j) {
[17:55:29]	                                            ^
[17:55:30]	[ 57%] Building CXX object paddle/fluid/operators/CMakeFiles/pool_op.dir/pool_op.cc.o
[17:55:30]	/paddle/paddle/fluid/operators/detection_map_op.h: In instantiation of 'void paddle::operators::DetectionMAPOpKernel<Place, T>::GetBoxes(const paddle::framework::LoDTensor&, const paddle::framework::LoDTensor&, std::vector<std::map<int, std::vector<paddle::operators::DetectionMAPOpKernel<Place, T>::Box> > >&, std::vector<std::map<int, std::vector<std::pair<T, paddle::operators::DetectionMAPOpKernel<Place, T>::Box> > > >&) const [with Place = paddle::platform::CPUPlace; T = float]':
[17:55:30]	/paddle/paddle/fluid/operators/detection_map_op.h:84:13:   required from 'void paddle::operators::DetectionMAPOpKernel<Place, T>::Compute(const paddle::framework::ExecutionContext&) const [with Place = paddle::platform::CPUPlace; T = float]'
[17:55:30]	/paddle/paddle/fluid/operators/detection_map_op.cc:184:67:   required from here
[17:55:30]	/paddle/paddle/fluid/operators/detection_map_op.h:154:38: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[17:55:30]	       for (int i = label_index[n]; i < label_index[n + 1]; ++i) {
[17:55:30]	                                      ^
[17:55:30]	[ 57%] Building CXX object paddle/fluid/operators/CMakeFiles/box_coder_op.dir/box_coder_op.cc.o
[17:55:30]	/paddle/paddle/fluid/operators/detection_map_op.h:170:39: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[17:55:30]	       for (int i = detect_index[n]; i < detect_index[n + 1]; ++i) {
[17:55:30]	                                       ^
[17:55:30]	[ 57%] Building CXX object paddle/fluid/operators/CMakeFiles/lstm_op.dir/lstm_op.cc.o
[17:55:30]	/paddle/paddle/fluid/operators/detection_map_op.h: In instantiation of 'paddle::operators::DetectionMAPOpKernel<Place, T>::GetInputPos(const paddle::framework::Tensor&, const paddle::framework::LoDTensor&, const paddle::framework::LoDTensor&, std::map<int, int>&, std::map<int, std::vector<std::pair<T, int> > >&, std::map<int, std::vector<std::pair<T, int> > >&) const::<lambda(const paddle::framework::LoDTensor&, std::map<int, std::vector<std::pair<T, int> > >&)> [with Place = paddle::platform::CPUPlace; T = float]':
[17:55:30]	/paddle/paddle/fluid/operators/detection_map_op.h:268:21:   required from 'struct paddle::operators::DetectionMAPOpKernel<Place, T>::GetInputPos(const paddle::framework::Tensor&, const paddle::framework::LoDTensor&, const paddle::framework::LoDTensor&, std::map<int, int>&, std::map<int, std::vector<std::pair<T, int> > >&, std::map<int, std::vector<std::pair<T, int> > >&) const [with Place = paddle::platform::CPUPlace; T = float]::<lambda(const class paddle::framework::LoDTensor&, class std::map<int, std::vector<std::pair<float, int>, std::allocator<std::pair<float, int> > >, std::less<int>, std::allocator<std::pair<const int, std::vector<std::pair<float, int>, std::allocator<std::pair<float, int> > > > > >&)>'
[17:55:30]	/paddle/paddle/fluid/operators/detection_map_op.h:280:5:   required from 'void paddle::operators::DetectionMAPOpKernel<Place, T>::GetInputPos(const paddle::framework::Tensor&, const paddle::framework::LoDTensor&, const paddle::framework::LoDTensor&, std::map<int, int>&, std::map<int, std::vector<std::pair<T, int> > >&, std::map<int, std::vector<std::pair<T, int> > >&) const [with Place = paddle::platform::CPUPlace; T = float]'
[17:55:30]	/paddle/paddle/fluid/operators/detection_map_op.h:91:18:   required from 'void paddle::operators::DetectionMAPOpKernel<Place, T>::Compute(const paddle::framework::ExecutionContext&) const [with Place = paddle::platform::CPUPlace; T = float]'
[17:55:30]	/paddle/paddle/fluid/operators/detection_map_op.cc:184:67:   required from here
[17:55:30]	/paddle/paddle/fluid/operators/detection_map_op.h:272:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[17:55:30]	       for (int i = 0; i < pos_data_lod.size(); ++i) {
[17:55:30]	                         ^
[17:55:30]	/paddle/paddle/fluid/operators/detection_map_op.h:273:44: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[17:55:30]	         for (int j = pos_data_lod[0][i]; j < pos_data_lod[0][i + 1]; ++j) {
[17:55:30]	    
```

--- a/paddle/fluid/operators/detection_map_op.h
+++ b/paddle/fluid/operators/detection_map_op.h
@@ -151,7 +151,7 @@ class DetectionMAPOpKernel : public framework::OpKernel<T> {
 
     for (int n = 0; n < batch_size; ++n) {
       std::map<int, std::vector<Box>> boxes;
-      for (int i = label_index[n]; i < label_index[n + 1]; ++i) {
+      for (size_t i = label_index[n]; i < label_index[n + 1]; ++i) {
         Box box(labels(i, 2), labels(i, 3), labels(i, 4), labels(i, 5));
         int label = labels(i, 0);
         auto is_difficult = labels(i, 1);
@@ -167,7 +167,7 @@ class DetectionMAPOpKernel : public framework::OpKernel<T> {
     auto detect_index = detect_lod[0];
     for (int n = 0; n < batch_size; ++n) {
       std::map<int, std::vector<std::pair<T, Box>>> boxes;
-      for (int i = detect_index[n]; i < detect_index[n + 1]; ++i) {
+      for (size_t i = detect_index[n]; i < detect_index[n + 1]; ++i) {
         Box box(detect(i, 2), detect(i, 3), detect(i, 4), detect(i, 5));
         int label = detect(i, 0);
         auto score = detect(i, 1);
@@ -269,8 +269,8 @@ class DetectionMAPOpKernel : public framework::OpKernel<T> {
                       std::map<int, std::vector<std::pair<T, int>>>& pos) {
       const T* pos_data = pos_tensor.data<T>();
       auto pos_data_lod = pos_tensor.lod();
-      for (int i = 0; i < pos_data_lod.size(); ++i) {
-        for (int j = pos_data_lod[0][i]; j < pos_data_lod[0][i + 1]; ++j) {
+      for (size_t i = 0; i < pos_data_lod.size(); ++i) {
+        for (size_t j = pos_data_lod[0][i]; j < pos_data_lod[0][i + 1]; ++j) {
           T score = pos_data[j * 2];
           int flag = 1;
           if (pos_data[j * 2 + 1] < kEPS) flag = 0;

